WalletWasabi.Gui/Global.cs

```diff
@@ -164,7 +160,9 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 					}
 
 					Tor.Http.TorHttpClient torHttpClient = BackendHttpClientFactory.NewTorHttpClient(Mode.DefaultCircuit);
-					HostedServices.Register<TorMonitor>(new TorMonitor(period: TimeSpan.FromSeconds(3), fallbackBackendUri: Config.GetFallbackBackendUri(), torHttpClient, TorManager), nameof(TorMonitor));
+					TorMonitor torMonitor = new(period: TimeSpan.FromSeconds(3), fallbackBackendUri: Config.GetFallbackBackendUri(), torHttpClient, TorManager);
+					_ = torMonitor.StartBootstrapMonitorAsync(cancel);
```

Comment:
I think this should go inside the TorMonitor and awaited properly by the `IHostedService` pattern implementation of it. In other words:

Shouldn't this be managed by the `HostedServices`? One of the points `TorMonitor` being a hosted service, managed by the `HostedServices` is to not have to do these initializations manually.

Reply:
btw: When I started with this PR, I actually wanted to remove the "periodic part" of TorMonitor and replace it with "reacting to Tor Control events" but I'm still not entirely sure how to do that correctly. That would make `TorMonitor` less hybrid I guess.

> Shouldn't this be managed by the `HostedServices`? One of the points `TorMonitor` being a hosted service, managed by the `HostedServices` is to not have to do these initializations manually.

I'll have a look whether I can modify it so that it's a proper `HostedService`.

Reply:
> I think this should go inside the TorMonitor and awaited properly by the `IHostedService` pattern implementation of it. In other words:

I have just modified it in 3a54d8ab400e0f5a5bc941115995dbdf5d6502ee and it looks certainly better.